velox/exec/Window.cpp

```diff
@@ -376,15 +419,20 @@ void Window::callApplyForPartitionRows(
     rawPeerEnds[j] = peerEndRow_ - 1 - firstPartitionRow;
 
     for (auto w = 0; w < numFuncs; w++) {
-      auto frameEndPoints = findFrameEndPoints(
-          w,
-          firstPartitionRow,
-          lastPartitionRow,
-          peerStartRow_,
-          peerEndRow_ - 1,
-          i);
-      rawFrameStartBuffers[w][j] = frameEndPoints.first - firstPartitionRow;
-      rawFrameEndBuffers[w][j] = frameEndPoints.second - firstPartitionRow;
+      rawFrameStartBuffers[w][j] = windowFrames_[w].findFrameBoundsStart(
+                                       windowFrames_[w].type,
+                                       w,
+                                       firstPartitionRow,
+                                       lastPartitionRow,
+                                       firstPartitionRow + rawPeerStarts[j]) -
```

Comment:
I feel we shouldn't overload this parameter this way, instead send the peer values separately. This has different logic for rows and range.